internals/src/Tizen.Peripheral/Tizen.Peripheral/Gpio.cs

```diff
@@ -0,0 +1,249 @@
+    /// </summary>
+    /// <privilege>http://tizen.org/privilege/peripheralio</privilege>
+    public class Gpio : IDisposable
+    {
+
+        private GpioChangePolarity _polarityType;
+        private NativeGpio.InterruptedEventCallback _interruptedEventCallback;
+
+        /// <summary>
+        /// Native handle to Gpio.
+        /// </summary>
+        private IntPtr _handle;
+        private bool _disposed = false;
+
+        /// <summary>
+        /// Invoked when Gpio pin was interrupted.
+        /// </summary>
+        public event EventHandler<PinUpdatedEventArgs> ValueChanged;
+
+        /// <summary>
+        /// Opens a GPIO pin.
+        /// </summary>
+        /// <param name="pinNumber">The GPIO pin number.</param>
+        /// <param name="mode">GPIO direction.</param>
+        public Gpio(int pinNumber, GpioPinDriveMode mode)
+        {
+            var ret = NativeGpio.Open(pinNumber, out IntPtr handle);
+            if (ret != ErrorCode.None)
+                throw ExceptionFactory.CreateException(ret);
+
+            _handle = handle;
+            PinNumber = pinNumber;
+            try
+            {
+                switch (mode)
+                {
+                    case GpioPinDriveMode.Input:
+                        ret = NativeGpio.SetEdgeMode(handle, NativeGpio.EdgeType.Both);
+                        if (ret != ErrorCode.None)
+                            throw ExceptionFactory.CreateException(ret);
+
+                        ret = NativeGpio.SetDirection(handle, NativeGpio.Direction.In);
+                        if (ret != ErrorCode.None)
+                            throw ExceptionFactory.CreateException(ret);
+                        SetIntteruptedCallback();
+                        break;
+                    case GpioPinDriveMode.OutputInitiallyLow:
+                        ret = NativeGpio.SetDirection(handle, NativeGpio.Direction.OutLow);
+                        if (ret != ErrorCode.None)
+                            throw ExceptionFactory.CreateException(ret);
+                        break;
+                    case GpioPinDriveMode.OutputInitiallyHigh:
+                        ret = NativeGpio.SetDirection(handle, NativeGpio.Direction.OutHigh);
+                        if (ret != ErrorCode.None)
+                            throw ExceptionFactory.CreateException(ret);
+                        break;
+                }
+            } catch(Exception e) {
+                Dispose();
+                throw e;
+            }
+        }
+
+
+        /// <summary>
+        /// Closes the GPIO pin.
+        /// </summary>
+        ~Gpio()
+        {
+            Dispose(false);
+        }
+
+        private void SetIntteruptedCallback()
+        {
+            _interruptedEventCallback = OnInterrupted;
+            var ret = NativeGpio.SetInterruptedCb(_handle, _interruptedEventCallback, IntPtr.Zero);
+            if (ret != ErrorCode.None)
+            {
+                Dispose();
```

Comment:
No need to call dispose, because it only called in constructor and it wrapped with try/catch
```suggestion

```